bot/exts/filters/filtering.py

```diff
@@ -207,6 +207,12 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    @Cog.listener()
+    async def on_voice_state_update(self, member: Member, before: VoiceState, after: VoiceState) -> None:
+        """Checks for bad words in usernames when users join or change voice channels."""
```

Comment:
This docstring isn't accurate. The `if after.channel and not before.channel` won't trigger when a member changes voice channels. It only triggers when a user joins a voice channel and wasn't in a voice channel previously.

If you want the described behaviour, you'd probably want `if after.channel and after.channel != before.channel`, although I personally think the current behaviour is better.

Reply:
good point, I'll clarify in the docstring

"""Checks for bad words in usernames when users initially joins a voice channel."""

Reply:
The "joins" should technically be singular, but I think that's too small to be worth fixing. Thanks for updating :+1: